db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -461,6 +464,11 @@ public GenericLanguageConnectionContext(
             // no op, use default value 6
         }
 
+        String costModelString = PropertyUtil.getCachedDatabaseProperty(this, Property.COST_MODEL);
```

Comment:
Why set it here and not in CompilerContext like the other global database properties?

Reply:
Logic is moved to `GenericStatement` and `CompilerContext`.